### PR TITLE
Refactor PaintAddImageAsParent overloads

### PR DESCRIPTION
**Pull request by ZehMatt**
99% of them were done with regex so I'm pretty confident I didn't break anything, for the other 1% I relied on compiler errors and fixed them by hand to be sure.

### CODE REVIEW

File: src/openrct2/paint/tile_element/Paint.LargeScenery.cpp [hunk 1]
```diff
@@ -289,8 +289,7 @@ void PaintLargeScenery(paint_session* session, uint8_t direction, uint16_t heigh
     boxlength.x = s98E3C4[esi].length.x;
     boxlength.y = s98E3C4[esi].length.y;
     boxlength.z = boxlengthZ;
-    PaintAddImageAsParent(
-        session, image_id, 0, 0, boxlength.x, boxlength.y, boxlengthZ, height, boxoffset.x, boxoffset.y, boxoffset.z);
+    PaintAddImageAsParent(session, image_id, { 0, 0, height }, { boxlength.x, boxlength.y, boxlengthZ }, boxoffset);
```
Comment (Gymnasiast) [inline src/openrct2/paint/tile_element/Paint.LargeScenery.cpp:292]: The line immediately above this already creates a CoordsXYZ for boxlength.

```suggestion
    PaintAddImageAsParent(session, image_id, { 0, 0, height }, boxLength, boxoffset);
```

File: src/openrct2/ride/thrill/MagicCarpet.cpp [hunk 2]
```diff
@@ -85,8 +85,8 @@ static void paint_magic_carpet_frame(
     if (plane == PLANE_BACK)
     {
         PaintAddImageAsParent(
-            session, imageId, static_cast<int8_t>(offset.x), static_cast<int8_t>(offset.y), bbSize.x, bbSize.y, 127, offset.z,
-            bbOffset.x, bbOffset.y, bbOffset.z);
+            session, imageId, { static_cast<int8_t>(offset.x), static_cast<int8_t>(offset.y), offset.z },
+            { bbSize.x, bbSize.y, 127 }, { bbOffset.x, bbOffset.y, bbOffset.z });
```
Comment (Gymnasiast) [inline src/openrct2/ride/thrill/MagicCarpet.cpp:89]:
```suggestion
            { bbSize.x, bbSize.y, 127 }, bbOffset);
```

File: src/openrct2/ride/coaster/WoodenRollerCoaster.cpp [hunk 3]
```diff
@@ -428,8 +428,9 @@ static void wooden_rc_track_paint_bb(paint_session* session, const sprite_bb_2*
     uint32_t railsImageId = bb->sprite_id_b | wooden_rc_get_rails_colour(session);
 
     PaintAddImageAsParent(
-        session, imageId, static_cast<int8_t>(bb->offset.x), static_cast<int8_t>(bb->offset.y), bb->bb_size.x, bb->bb_size.y,
-        static_cast<int8_t>(bb->bb_size.z), height + bb->offset.z, bb->bb_offset.x, bb->bb_offset.y, height + bb->bb_offset.z);
+        session, imageId, { static_cast<int8_t>(bb->offset.x), static_cast<int8_t>(bb->offset.y), height + bb->offset.z },
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/WoodenRollerCoaster.cpp:431]: Do we still need these static casts? I bet they were once needed to handle signed/unsignedness, but have been obsolete for a long time.

Comment (ZehMatt) [reply src/openrct2/ride/coaster/WoodenRollerCoaster.cpp:431]: We don't I missed them

**Review by Gymnasiast — COMMENTED** [review]
Looked through all the code and found nothing that was wrong, just a few places where more cleanup was possible.

I do want to test the code before I approve it, though.

File: src/openrct2/paint/Supports.cpp [hunk 4]
```diff
@@ -607,8 +603,8 @@ bool wooden_a_supports_paint_setup(
             if (byte_97B23C[special].var_6 == 0 || session->WoodenSupportsPrependTo == nullptr)
             {
                 PaintAddImageAsParent(
-                    session, imageId, 0, 0, bBox.length.x, bBox.length.y, bBox.length.z, z, bBox.offset.x, bBox.offset.y,
-                    bBox.offset.z + z);
+                    session, imageId, { 0, 0, z }, { bBox.length.x, bBox.length.y, bBox.length.z },
```
Comment (duncanspumpkin) [inline src/openrct2/paint/Supports.cpp:606]: Can remove construction of temporary

File: src/openrct2/ride/TrackPaint.cpp [hunk 5]
```diff
@@ -1644,8 +1638,8 @@ void track_paint_util_right_quarter_turn_3_tiles_paint(
     CoordsXYZ boundsOffset = (boundsOffsets == nullptr ? CoordsXYZ(offset, 0) : boundsOffsets[direction][index]);
 
     PaintAddImageAsParent(
-        session, imageId, static_cast<int8_t>(offset.x), static_cast<int8_t>(offset.y), boundsLength.x, boundsLength.y,
-        thickness, height, boundsOffset.x, boundsOffset.y, height + boundsOffset.z);
+        session, imageId, { static_cast<int8_t>(offset.x), static_cast<int8_t>(offset.y), height },
```
Comment (duncanspumpkin) [inline src/openrct2/ride/TrackPaint.cpp:1641]: casts not required

**Review by duncanspumpkin — APPROVED** [review]
All changes look fine.

**Review by Gymnasiast — APPROVED** [review]
Appears to work correctly.

**Pull request merged**